fhirstore/search_engine.py

```diff
@@ -1,21 +1,58 @@
-from os import getenv
+import os
+import json
+import logging
 from yarl import URL
 
+from fhirstore.errors import SearchEngineError
+
 from fhirpath.connectors.factory.es import ElasticsearchConnection
 from fhirpath.engine.es import ElasticsearchEngine as BaseEngine
-from fhirpath.engine import dialect_factory
-from fhirpath.enums import FHIR_VERSION
-from fhirpath_helpers.elasticsearch.mapping import generate_mappings
+from fhirpath.engine import dialect_factory, EngineResultRow
 
-FHIR_API_URL = getenv("FHIR_API_URL", "https://arkhn.com")
+FHIR_API_URL = os.getenv("FHIR_API_URL", "https://arkhn.com")
+ELASTIC_MAPPING_FILES_DIR = os.getenv("ELASTIC_MAPPING_FILES_DIR")
 
 
 class ElasticSearchEngine(BaseEngine):
-    def __init__(self, es_client, fhir_release=FHIR_VERSION.R4):
+
+    es_reference_analyzer = "fhir_reference_analyzer"
+    es_token_normalizer = "fhir_normalizer"
+
+    def __init__(self, fhir_release, es_client, es_index):
         super().__init__(
             fhir_release, lambda x: ElasticsearchConnection(es_client), dialect_factory
         )
-        self.mappings = generate_mappings(FHIR_VERSION.R4.name)
+        self.es_index = es_index
+        self.mappings = {}
+
+        # load the ES mappings from static files
+        if ELASTIC_MAPPING_FILES_DIR:
+            logging.info(f"Loading ES index from {ELASTIC_MAPPING_FILES_DIR}...")
+            try:
+                mapping_files = os.listdir(ELASTIC_MAPPING_FILES_DIR)
```

Comment:
I think using `glob` would slightly clean the code here.
You can do `glob(ELASTIC_MAPPING_FILES_DIR/**.json)` with this package to get all the files in `ELASTIC_MAPPING_FILES_DIR` (and no need for the `os.path.join`).

Reply:
cool, thanks done